pkg/util/k8sconfig/k8s_client_config.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2024 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package k8sconfig helps to build kubernetes configurations.
+package k8sconfig
+
+import (
+	restclient "k8s.io/client-go/rest"
+	clientcmd "k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog"
+)
+
+// InClusterBuildConfig is a helper function that builds configs by trying the InClusterConfig().
+// If InClusterConfig is unsuccessful, it falls back to BuildConfigFromFlags.
+func InClusterBuildConfig(masterURL, kubeconfigPath string) (*restclient.Config, error) {
```

Comment:
```suggestion
func InClusterBuildConfig(kubeconfigPath string) (*restclient.Config, error) {
```

Since we never use a `masterURL` other than "" ?